### ISSUE TITLE
Fix noun article on annotation press

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-iOS#302]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-iOS/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-iOS' [Code of Conduct](https://github.com/scribe-org/Scribe-iOS/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

Amazing feature idea that came up via showing Scribe to a coworker 🎉 It would be great if we added a feature where pressing the noun annotation would check to see if there’s an article before it and if so if it’s the incorrect article. If it’s not correct then the word and the article can then be replaced with the correct one and the word after it 😊

Things to consider:
- We’d need to make sure the key gets the shadow, which I believe comes from the active state regardless
- For words that are both nouns and prepositions the article correction would first be triggered and then if the article is correct we could do preposition pronoun selections
- It’d be good if the replacement was matched with an indicator to the user, with the current working thought being that we could display a checkmark icon on the annotation for a moment before reverting back to the annotation

This feature would definitely be worth a release for v2.4 🚀 

### Contribution

I’ll start working on this soon and would be happy to get some support :)

**Comment by andrewtavis** [comment]
@SaurabhJamadagni, @wkyoshida: tagging to bring this to your two’s attention as this would be really great :) One of those can’t believe we didn’t think of this features 😊 Consistently during demos I’m telling people that they’d at least know that they need to go back and change the article if it’s wrong, so this would be huge!

**Comment by andrewtavis** [comment]
Something to consider for this is that we also need to factor for words that could have more than one gender — for instance the class `Schild` in German — such that the user would first be prompted to select the correct gender. Beyond this if there is more than one case possible then we don't necessarily want to show the options for other genders in the normal selection as that would allow them to insert an incorrect value. In cases where there is more than one gender possible and it's after a preposition that has more than one case, then they should first select the gender and then the case.

**Comment by andrewtavis** [comment]
For words that have more than one gender we should set it up so that clicking the annotation brings up a display to check/click the right one. This would then allow us to later add functionality where the user could check if they've used the right article given their meaning via the importing of the senses of words such that we could also display the sense as a definition to the user.

**Comment by SaurabhJamadagni** [comment]
Hey @andrewtavis, sounds like a great feature idea! I am kinda lost on the intricacies though. Could you explain them to me when we have our call? 😅

**Comment by andrewtavis** [comment]
Yes yes, definitely can explain more, @SaurabhJamadagni 😊 As you can see was one of those ideas that kind of snowballed after we originally came up with it 😅😊😊

I’ll plan on getting the emulator out for the call to explain what would happen :)

**Comment by andrewtavis** [comment]
Based on a comment from @SaurabhJamadagni, the checkmark should be displayed even if the article is correct. We decided with @wkyoshida to do the checkmark if it has been corrected, and a thumbs up if it was correct in the first place. These symbols would come from SFSymbols :)

**Comment by andrewtavis** [comment]
Something to potentially add to this would be a quick animation within the command area that shows the correct article and the noun together and colored with the given gender color. An example:

- The user types "der Buch" on a German Scribe keyboard (should be "das")
- The users sees the `N` annotation and realizes that "der" is wrong
- The user presses the Scribe key and the article is replaced
- As confirmation of the command, the command bar is shown for a second with "das Buch" as the text on it
- The command bar's background color is green for neutral nouns, and the text is the same color as the annotation text color

I think that this would be a good way of doing this such that we're reinforcing the correct gender with the quick fix command :)
